packages/extension-dapp/src/index.ts

```diff
@@ -178,12 +178,12 @@ export async function web3FromSource (source: string): Promise<InjectedExtension
 }
 
 // find a specific provider based on an address
-export async function web3FromAddress (address: string): Promise<InjectedExtension> {
+export async function web3FromAddress (address: string, { ss58Format }: Web3AccountsOptions = {}): Promise<InjectedExtension> {
   if (!web3EnablePromise) {
     return throwError('web3FromAddress');
   }
 
-  const accounts = await web3Accounts();
+  const accounts = await web3Accounts({ ss58Format });
```

Comment:
Since the signer keypairs operate on the actual raw (non-ss58), the actual solution is actually the next line.

Instad of simply doing `account.address === address` it should -

- decode the address into hex
- do the compare agains the list in hex

That way it always works on the raw representation, no matter which ss58 is passed in.

Reply:
... and instead of the `u8aToHex`, we can just decode and use `u8aEq(decodedAddress, decodeAddress(account.address))` (which is more optimal since there is no u8a -> hex involved)

Reply:
```javascript
const allAccounts = await web3Accounts({ ss58Format: 42 });
const injector = await web3FromAddress(allAccounts[0].address);
```
Error: web3FromAddress: Unable to find injected 525BJ6fSuXDiTTHH8ubfpEpZRR7FZgupSLy6d1hbZZ3K4ZxE"

Reply:
As above, the check on line 107 is the right place to adjust.